src/main/java/spoon/reflect/visitor/PrintingContext.java

```diff
@@ -26,17 +27,22 @@
 
 public class PrintingContext {
 
-	private long NO_TYPE_DECL 			= 1 << 0;
+	private long NEXT_FOR_VARIABLE 			= 1 << 0;
 	private long IGNORE_GENERICS 		= 1 << 1;
 	private long SKIP_ARRAY 			= 1 << 2;
 	private long IGNORE_STATIC_ACCESS   = 1 << 3;
 	private long IGNORE_ENCLOSING_CLASS = 1 << 4;
 	private long FORCE_WILDCARD_GENERICS = 1 << 5;
+	private long FIRST_FOR_VARIABLE 	= 1 << 6;
 
 	private long state;
+	private CtStatement statement;
 
-	public boolean noTypeDecl() {
```

Comment:
instead of removing it, couldn't you just deprecate it with checking the flag: `NEXT_FOR_VARIABLE`? I imagine the semantic of the flag did not really change did it?

Reply:
semantic of the flag is same. It just have better name. OK. So I will deprecate old name and create new name with same internal property